credentials/xds/xds.go

```diff
@@ -173,13 +204,14 @@ func (hi *HandshakeInfo) makeTLSConfig(ctx context.Context) (*tls.Config, error)
 	// includes hostname verification) or none. We are forced to go with the
 	// latter and perform the normal cert validation ourselves.
 	cfg := &tls.Config{InsecureSkipVerify: true}
-	if rootProv != nil {
-		km, err := rootProv.KeyMaterial(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("xds: fetching trusted roots from CertificateProvider failed: %v", err)
-		}
-		cfg.RootCAs = km.Roots
+
```

Comment:
Can I know why we don't need to check `rootProv` is nil here? Is it because we always make a call to `validate` and then call `make*SideTLSConfig` here? Would it be better if we merge `validate` and `make*SideTLSConfig` into one single function?

Reply:
Thanks for the suggestion. Incorporated it and got rid of the `validate()` function.